### ISSUE TITLE
Celery dies with no error message

### ISSUE THREAD

**Issue by darkpixel** [celery/celery#2742]
Celery is dying with no error message:

```
12:47:29 [aaron@praxis:~/code/uitintranet] [uitintranet] master* ± DATABASE_URL=postgres://uitintranet:SNIP@tetrad.OURDOMAIN.com/uitintranet PRODUCTION=True REDIS_URL=redis://tetrad.OURDOMAIN.com:6379 BROKER_PASSWORD=SNIP BROKER_PATH=/host_uitintranet BROKER_SCHEME=amqp BROKER_URL=amqp://${BROKER_USERNAME}:${BROKER_PASSWORD}@${BROKER_PORT_5672_TCP_ADDR}:${BROKER_PORT_15672_TCP_PORT}/ BROKER_USERNAME=uitintranet python manage.py celery worker
 
 -------------- celery@praxis v3.1.18 (Cipater)
---- **** ----- 
--- * ***  * -- Linux-3.13.0-37-generic-x86_64-with-LinuxMint-17.1-rebecca
-- * - **** --- 
- ** ---------- [config]
- ** ---------- .> app:         uitintranet:0x7f4fa978df90
- ** ---------- .> transport:   redis://tetrad.OURDOMAIN.com:6379//
- ** ---------- .> results:     redis://tetrad.OURDOMAIN.com:6379
- *** --- * --- .> concurrency: 16 (prefork)
-- ******* ---- 
--- ***** ----- [queues]
 -------------- .> celery           exchange=celery(direct) key=celery
                

[2015-08-03 12:47:33,106: WARNING/MainProcess] celery@praxis ready.
12:47:35 [aaron@praxis:~/code/uitintranet] [uitintranet] master* ± 
```
It displays the config settings, says 'ready', then hangs for a few seconds, and returns to the command prompt.

Here's with --loglevel DEBUG:

```
[2015-08-03 12:50:49,003: DEBUG/MainProcess] | Worker: Preparing bootsteps.
[2015-08-03 12:50:49,007: DEBUG/MainProcess] | Worker: Building graph...
[2015-08-03 12:50:49,008: DEBUG/MainProcess] | Worker: New boot order: {Beat, StateDB, Timer, Hub, Queues (intra), Pool, Autoreloader, Autoscaler, Consumer}
[2015-08-03 12:50:49,017: DEBUG/MainProcess] | Consumer: Preparing bootsteps.
[2015-08-03 12:50:49,018: DEBUG/MainProcess] | Consumer: Building graph...
[2015-08-03 12:50:49,031: DEBUG/MainProcess] | Consumer: New boot order: {Connection, Events, Heart, Mingle, Tasks, Control, Agent, Gossip, event loop}
 
 -------------- celery@praxis v3.1.18 (Cipater)
---- **** ----- 
--- * ***  * -- Linux-3.13.0-37-generic-x86_64-with-LinuxMint-17.1-rebecca
-- * - **** --- 
- ** ---------- [config]
- ** ---------- .> app:         uitintranet:0x7fd984165f90
- ** ---------- .> transport:   redis://tetrad.OURDOMAIN.com:6379//
- ** ---------- .> results:     redis://tetrad.OURDOMAIN.com:6379
- *** --- * --- .> concurrency: 16 (prefork)
-- ******* ---- 
--- ***** ----- [queues]
 -------------- .> celery           exchange=celery(direct) key=celery
                

[tasks]
  . celery.backend_cleanup
  . celery.chain
  . celery.chord
  . celery.chord_unlock
  . celery.chunks
  . celery.group
  . celery.map
  . celery.starmap
  . celery_haystack.tasks.CeleryHaystackSignalHandler
  . celery_haystack.tasks.CeleryHaystackUpdateIndex
  . cw.tasks.close_toggl_projects
  . cw.tasks.daily_toggl_mentions
  . cw.tasks.gpsify_ticket
  . cw.tasks.gpsify_tickets
  . cw.tasks.mention_latest_tickets
  . cw.tasks.sync_clients_to_toggl
  . cw.tasks.sync_cw_ticket_to_rt
  . cw.tasks.sync_reopened_tickets_to_toggl
  . cw.tasks.sync_ticket_to_toggl
  . cw.tasks.troll_rt
  . cw.tasks.update_stats
  . cw.tasks.update_ticket_details
  . cw.tasks.update_ticket_statuses
  . cw.tasks.update_toggl_stats
  . djcelery_email_send
  . gfi.tasks.update_eol_stats
  . intranet.tasks.backup_router
  . intranet.tasks.backup_routers
  . intranet.tasks.send_to_slack
  . intranet.tasks.update_stat
  . rt4.tasks.add_rt_comment
  . rt4.tasks.get_rt_by_cwid
  . rt4.tasks.get_rt_ticket
  . rt4.tasks.map_cw_status_to_rt
  . rt4.tasks.update_rt_ticket
  . smiles.tasks.backup_router
  . smiles.tasks.backup_routers
  . smiles.tasks.create_office_in_ad
  . smiles.tasks.gather_nas_master_keys
  . smiles.tasks.gather_nas_master_keys_all
  . smiles.tasks.gather_offsite_snapshots
  . smiles.tasks.gather_pool_stat
  . smiles.tasks.gather_pool_stats
  . smiles.tasks.gather_serial
  . smiles.tasks.gather_serials
  . smiles.tasks.gather_snapshot
  . smiles.tasks.gather_snapshots
  . smiles.tasks.unlock_pool
  . smiles.tasks.update_nas_status
  . smiles.tasks.update_nas_status_all_offices
  . uitintranet.celery.debug_task

[2015-08-03 12:50:49,038: DEBUG/MainProcess] | Worker: Starting Hub
[2015-08-03 12:50:49,038: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:49,039: DEBUG/MainProcess] | Worker: Starting Pool
[2015-08-03 12:50:49,100: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:49,101: DEBUG/MainProcess] | Worker: Starting Consumer
[2015-08-03 12:50:49,101: DEBUG/MainProcess] | Consumer: Starting Connection
[2015-08-03 12:50:49,119: INFO/MainProcess] Connected to redis://tetrad.OURDOMAIN.com:6379//
[2015-08-03 12:50:49,119: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:49,119: DEBUG/MainProcess] | Consumer: Starting Events
[2015-08-03 12:50:49,134: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:49,134: DEBUG/MainProcess] | Consumer: Starting Heart
[2015-08-03 12:50:49,136: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:49,136: DEBUG/MainProcess] | Consumer: Starting Mingle
[2015-08-03 12:50:49,136: INFO/MainProcess] mingle: searching for neighbors
[2015-08-03 12:50:50,292: INFO/MainProcess] mingle: sync with 1 nodes
[2015-08-03 12:50:50,319: INFO/MainProcess] mingle: sync complete
[2015-08-03 12:50:50,319: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:50,320: DEBUG/MainProcess] | Consumer: Starting Tasks
[2015-08-03 12:50:50,399: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:50,399: DEBUG/MainProcess] | Consumer: Starting Control
[2015-08-03 12:50:50,404: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:50,405: DEBUG/MainProcess] | Consumer: Starting Gossip
[2015-08-03 12:50:50,410: DEBUG/MainProcess] ^-- substep ok
[2015-08-03 12:50:50,410: DEBUG/MainProcess] | Consumer: Starting event loop
[2015-08-03 12:50:50,410: WARNING/MainProcess] celery@praxis ready.
[2015-08-03 12:50:50,411: DEBUG/MainProcess] | Worker: Hub.register Pool...
[2015-08-03 12:50:51,134: DEBUG/MainProcess] | Worker: Closing Hub...
[2015-08-03 12:50:51,134: DEBUG/MainProcess] | Worker: Closing Pool...
[2015-08-03 12:50:51,135: DEBUG/MainProcess] | Worker: Closing Consumer...
[2015-08-03 12:50:51,135: DEBUG/MainProcess] | Worker: Stopping Consumer...
[2015-08-03 12:50:51,135: DEBUG/MainProcess] | Consumer: Closing Connection...
[2015-08-03 12:50:51,135: DEBUG/MainProcess] | Consumer: Closing Events...
[2015-08-03 12:50:51,135: DEBUG/MainProcess] | Consumer: Closing Heart...
[2015-08-03 12:50:51,136: DEBUG/MainProcess] | Consumer: Closing Mingle...
[2015-08-03 12:50:51,136: DEBUG/MainProcess] | Consumer: Closing Tasks...
[2015-08-03 12:50:51,136: DEBUG/MainProcess] | Consumer: Closing Control...
[2015-08-03 12:50:51,136: DEBUG/MainProcess] | Consumer: Closing Gossip...
[2015-08-03 12:50:51,136: DEBUG/MainProcess] | Consumer: Closing event loop...
[2015-08-03 12:50:51,137: DEBUG/MainProcess] | Consumer: Stopping event loop...
[2015-08-03 12:50:51,137: DEBUG/MainProcess] | Consumer: Stopping Gossip...
[2015-08-03 12:50:51,141: DEBUG/MainProcess] | Consumer: Stopping Control...
[2015-08-03 12:50:51,146: DEBUG/MainProcess] | Consumer: Stopping Tasks...
[2015-08-03 12:50:51,146: DEBUG/MainProcess] Cancelling task consumer...
[2015-08-03 12:50:51,147: DEBUG/MainProcess] | Consumer: Stopping Mingle...
[2015-08-03 12:50:51,147: DEBUG/MainProcess] | Consumer: Stopping Heart...
[2015-08-03 12:50:51,148: DEBUG/MainProcess] | Consumer: Stopping Events...
[2015-08-03 12:50:51,149: DEBUG/MainProcess] | Consumer: Stopping Connection...
[2015-08-03 12:50:51,149: DEBUG/MainProcess] | Worker: Stopping Pool...
[2015-08-03 12:50:52,168: DEBUG/MainProcess] | Worker: Stopping Hub...
[2015-08-03 12:50:52,168: DEBUG/MainProcess] | Consumer: Shutdown Gossip...
[2015-08-03 12:50:52,168: DEBUG/MainProcess] | Consumer: Shutdown Control...
[2015-08-03 12:50:52,169: DEBUG/MainProcess] | Consumer: Shutdown Tasks...
[2015-08-03 12:50:52,169: DEBUG/MainProcess] Cancelling task consumer...
[2015-08-03 12:50:52,169: DEBUG/MainProcess] Closing consumer channel...
[2015-08-03 12:50:52,169: DEBUG/MainProcess] | Consumer: Shutdown Heart...
[2015-08-03 12:50:52,169: DEBUG/MainProcess] | Consumer: Shutdown Events...
[2015-08-03 12:50:52,170: DEBUG/MainProcess] | Consumer: Shutdown Connection...
[2015-08-03 12:50:52,171: DEBUG/MainProcess] removing tasks from inqueue until task handler finished
12:50:52 [aaron@praxis:~/code/uitintranet] [uitintranet] master* ± 
```

**Comment by ask** [comment]
The next thing you would expect to see after `Worker: Hub.register Pool...` is: `basic.qos: prefetch_count->80` so probably something happened between those two

**Comment by ask** [comment]
Maybe you can add some debugging here?

Add a try: ... except Exception as exc: print(exc) raise; around:
https://github.com/celery/celery/blob/master/celery/worker/consumer.py#L301-L305

and:
https://github.com/celery/celery/blob/master/celery/worker/consumer.py#L301-L305

**Comment by ask** [comment]
Closing due to inactivity

**Comment by ransage** [comment]
For others' general reference, I encountered a similar signature yesterday and it turned out that one of the VMs was unresponsive, reporting 100% cpu utilization to Nagios.  In my case, if I started a queue that was not running on the affected host, it would start just fine, but when starting the normal queues they would hang for 1-2 seconds when they should have been connecting to the cluster and then exit with status 0.  When the VM in question was hard-restarted, the Celery queues all behaved nominally.

As I understand the VM issue, this may not have been caused by a Celery problem at all, but that is where we first noticed it because processing queues were dropping off of flower.

**Comment by vovkd** [comment]
Hello!
Sadly, but i've got this message after updated celery 3 to celery 4 and updated kombu and billiard to latest versions (django 1.9) and I dont know what caused this. No erros, nothing to look into. 
```

celery worker -E --time-limit=300 --concurrency=1 -l debug -n w1 -Q impexp -A celeryapp:celery_app

 -------------- celery@w1 v4.0.2 (latentcall)
---- **** ----- 
--- * ***  * -- Linux-3.13.0-26-generic-i686-with-Ubuntu-14.04-trusty 2017-03-06 16:35:46
-- * - **** --- 
- ** ---------- [config]
- ** ---------- .> app:         celeryapp:0xb67c24ec
- ** ---------- .> transport:   amqp://guest:**@localhost:5672//
- ** ---------- .> results:     amqp://
- *** --- * --- .> concurrency: 1 (prefork)
-- ******* ---- .> task events: ON
--- ***** ----- 
 -------------- [queues]
                .> impexp           exchange=(direct) key=impexp
                

[tasks]
  . celery.accumulate
  . celery.backend_cleanup
  . celery.chain
  . celery.chord
  . celery.chord_unlock
  . celery.chunks
  . celery.group
  . celery.map
  . celery.starmap
  . impexp

[2017-03-06 16:35:47,286: DEBUG/MainProcess] removing tasks from inqueue until task handler finished
```

**Comment by wenbochang** [comment]
@vovkd did you find a workaround? i have exact same issue with cel 4.1 and django 1.10

**Comment by ransage** [comment]
We just encountered an issue very much like this again last week.  Our versions have been frozen on the VMs that were affected, so it was not an upgrade to 4 that affected us.

Again our issue seems to have had something to do with a bogus host clustered via RabbitMQ.  (I found my reply from last year when looking for info.)  Unlike last time, our host was not unresponsive generally, but was not clustering properly and reported itself as a disc (or ram) node when all of the other nodes were ram (or disc).

For us to solve the problem, we disabled the problematic node, we ensured configuration was consistent across remaining nodes, and then we still needed to shut down all RabbitMQ instances and restart them to get rabbit-cached info to go away - then the issue went away.

If I were to guess a conservative sequencing, it might be something like:

1. shut down all of the celery instances
2. shut down all of the RabbitMQ instances
3. verify RabbitMQ configurations
4. reboot
5. confirm that nodes came up as expected, such as with `sudo rabbitmqctl cluster_status`
6. restart celery nodes and see if they exit mysteriously or if they work

In our case, restarting the hosts without firsts shutting down all RabbitMQ instances did not flush cached RabbitMQ data (this was evidenced by the disabled node still appearing in cluster_status).

Hope that helps!
